pkg/kubectl/apply/element.go

```diff
@@ -419,3 +432,11 @@ func (e HasElementData) HasLocal() bool {
 func (e HasElementData) HasRemote() bool {
 	return e.remoteSet
 }
+
+// DetectConflictor defines conflict detection method on Element level.
+// Calling DetectConflict on an element will apply type specific detection method
+// A conflict is if the same field is specified in BOTH the recorded and the remote values
+// of an object, but does not match.
+type DetectConflictor interface {
```

Comment:
We should rename it to ConflictDetector. Its defining behavior for "detecting conflicts", doesn't have follow "xxxx-er" like syntax :) I know where you are coming from :)

We should improve the documentation saying:
// ConflictDetector defines the capability to detect conflict. An element can examine remote/recorded value to detect conflict. 
Also: we don't have to repeat definition of conflict everywhere. We can just define it at one place.